Sources/FoundationInternationalization/ICU/ICUShims.swift

```diff
@@ -0,0 +1,53 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if FOUNDATION_FRAMEWORK
+@_implementationOnly import FoundationICU
+#else
+package import FoundationICU
+#endif
+
+#if os(Windows)
+typealias EnumRawType = CInt
+#else
+typealias EnumRawType = CUnsignedInt
+#endif
+
+internal func _ucal_getDayOfWeekType(_ cal: UnsafePointer<UCalendar?>,
+                                     _ dayOfWeek: CInt) -> CInt {
+  var status: UErrorCode = U_ZERO_ERROR
+  let value: UCalendarWeekdayType =
+      ucal_getDayOfWeekType(cal,
+                            UCalendarDaysOfWeek(rawValue: EnumRawType(dayOfWeek)),
+                            &status)
+  return CInt(value.rawValue)
+}
+
+internal func _unum_setSymbol(_ formatter: UnsafeMutablePointer<UNumberFormat?>,
+                              _ symbol: CInt, _ value: UnsafePointer<UChar>?,
+                              _ length: CInt) {
+  var status: UErrorCode = U_ZERO_ERROR
+  unum_setSymbol(formatter, UNumberFormatSymbol(rawValue: EnumRawType(symbol)),
+                 value, length, &status)
+}
+
+extension UDateFormatField {
+  internal init(_ rawValue: CInt) {
+    self.init(rawValue: EnumRawType(rawValue))
+  }
+}
+
+extension UNumberFormatFields {
+  internal init(_ rawValue: CInt) {
+    self.init(rawValue: EnumRawType(rawValue))
+  }
+}
```

Comment:
If you'll permit me one nitpick - can we instead add these to the `ICU+Enums.swift` file, since I think that's actually the one meant to hold this kind of thing? I'm trying to keep these ICU extensions organized instead of ending up with a grab bag 'shims' file.

Reply:
Sure, I don't mind that at all.  I just didn't notice the file.